include/k4arecord/playback.hpp

```diff
@@ -72,6 +279,34 @@ class playback
         return m_handle != nullptr;
     }
 
+    /** Returns true if two playback objects refer to the same k4a_playback_t, false otherwise
+     */
+    bool operator==(const playback &other) const noexcept
+    {
+        return m_handle == other.m_handle;
+    }
+
+    /** Returns false if the playback object is valid, true otherwise
+     */
+    bool operator==(std::nullptr_t) const noexcept
+    {
+        return m_handle == nullptr;
+    }
+
+    /** Returns true if two playback objects wrap different k4a_playback_t instances, false otherwise
+     */
+    bool operator!=(const playback &other) const noexcept
+    {
+        return m_handle != other.m_handle;
+    }
+
+    /** Returns true if the playback object is valid, false otherwise
+     */
+    bool operator!=(std::nullptr_t) const noexcept
+    {
+        return m_handle != nullptr;
+    }
```

Comment:
I'm not sure these work with the semantics of k4a::playback - it's not a pseudo-pointer type like k4a::image, we delete the copy constructor so two playbacks should never be equal and we don't allow assigning nullptr to the playback, so allowing comparison to nullptr seems really weird

Reply:
It is the same, just not ref counted, exactly the same as 'device'. Seems like if we had the bool operator then we should allow for == != nullptr? no?

Reply:
The device class also doesn't have comparison operators for the same reason :)  The problem is you actually can't have 2 playbacks/records/devices wrapping the same underling C playback/record/device handles, so comparing to anything doesn't make sense.  We don't provide a copy constructor or copy operator= for those types because if we did, the second wrapper would segfault in its destructor because it's trying to free memory that's already been freed by the first one's destructor.  I suppose you could define just operator==(nullptr_t), but that's also super weird because it's not really behaving like a refcounted pointer in the same way a k4a::image is, we're just randomly using ==(nullptr_t) as an is_valid() function.  operator bool is idiomatic for "is this thing valid" even on non-pointerish types (e.g. std::ostream/std::istream do this), but they don't allow pointer type comparisons either

---
In reply to: [347614993](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/pull/912#discussion_r347614993) [](ancestors = 347614993)